Wabbajack/View Models/ModVM.cs

```diff
@@ -13,37 +13,39 @@ namespace Wabbajack
 {
     public class ModVM : ViewModel
     {
-        public string ModName { get; }
+        public IMetaState State { get; }
 
-        public string ModID { get; }
-
-        public string ModDescription { get; }
-
-        public string ModAuthor { get; }
-
-        public bool IsNSFW { get; }
-
-        public string ModURL { get; }
-        
+        public string URL { get; }
+        public string Name { get; }
+        public string Author { get; }
+        public string Version { get; }
         public string ImageURL { get; }
+        public bool IsNSFW { get; }
+        public string Description { get; }
 
         // Image isn't exposed as a direct property, but as an observable.
         // This acts as a caching mechanism, as interested parties will trigger it to be created,
         // and the cached image will automatically be released when the last interested party is gone.
         public IObservable<BitmapImage> ImageObservable { get; }
 
-        public ModVM(NexusDownloader.State m)
+        public ModVM(IMetaState state)
         {
-            ModName = NexusApiUtils.FixupSummary(m.ModName);
-            ModID = m.ModID;
-            ModDescription = NexusApiUtils.FixupSummary(m.Summary);
-            ModAuthor = NexusApiUtils.FixupSummary(m.Author);
-            IsNSFW = m.Adult;
-            ModURL = m.NexusURL;
-            ImageURL = m.SlideShowPic;
+            State = state;
+
+            URL = state.URL;
+            ImageURL = state.ImageURL;
+            IsNSFW = state.IsNSFW;
+            Version = state.Version;
+
+            var isNexus = state.GetType() == typeof(NexusDownloader.State);
+            
+            Name = isNexus ? NexusApiUtils.FixupSummary(state.Name) : state.Name;
```

Comment:
I think ModVM should be agnostic of whether it's nexus or not.  the IMetaState interface should be doing any fixup needed internally and exposing the correct name, i think.

Reply:
Just to clarify, I think while the `State` class implements `IMetaState` and has a `Name` property... NexusState can override it and provide the fixup itself internally.

Either by actually doing it to the `Name` property straight up, or alternatively, can do the fixup when `Name` is accessed through the `IMetaState.Name` member only, like this:
```
// Property still exists on the base class like normal
string Name { get; set; }

// Intercept and override the behavior for IMetaState.Name calls only
string IMetaState.Name
{
   get => NexusApiUtils.FixupSummary(this.Name);
   set => this.Name = value;
}
```